src/libraries/Microsoft.PowerFx.Interpreter/ParsedExpression.cs

```diff
@@ -84,7 +83,7 @@ internal static IExpressionEvaluator GetEvaluator(this CheckResult result, Stack
             var expr = new ParsedExpression(irResult.TopNode, irResult.RuleScopeSymbol, stackMarker, result.CultureInfo)
             {
                 _globals = globals,
-                _allSymbols = allSymbols,
+                _allSymbols = result.Symbols,

```

Comment:
This was the bug. Calling Symbols instead of AllSymbols would have caught it.  Hence demoting allSymbols to a private field to avoid misuse.